meta-client/src/main/java/org/apache/helix/metaclient/api/MetaClientInterface.java

```diff
@@ -143,7 +152,7 @@ public int getVersion() {
    * @eturn Return a list of children keys. Return direct child name only for hierarchical key
    *        space, return the whole sub key for non-hierarchical key space.
    */
-  List<String> getDirestChildrenKeys(final String key);
+  List<String> getDirectChildrenKeys(final String key);
```

Comment:
this is very confusing. for non-hierarchical, shouldn't it return NotImplemented ?

Reply:
TFTR. We want to keep the possibility that user may want to mimic hierarchical key space using a non-hierarchical KV store. (e.g. Helix explored the possibility of using etcd). In this case, user will pass a prefix as key the `getDirectChildrenKeys` returns all keys matches the prefix.

Reply:
Got it. So interface will still say, getChildren() but Etcd will implement using prefix search.. in that case, name is very confusing for real metadata client user. For example, kv store, will they use 'getchildren() or should use getprefixsearch() or equivalent.

Reply:
They will use `getchildren()` as there is no `getprefixsearch ()` in the interface.  We could add that for specific impl in the future.

Reply:
Based on offline discussion, let's keep the naming for now and may add more API when needed in future.